### ISSUE TITLE
can not recognize `del` tag without space

### ISSUE THREAD

**Issue by asjqkkkk** [dart-lang/markdown#300]
## markdown source

```
pas~~t~~
```
## parse code

```
    final content = 'pas~~t~~';
    final m.Document document = m.Document(
        extensionSet: m.ExtensionSet.gitHubFlavored,
        encodeHtml: false);
    final List<String> lines = content.split(RegExp(r'\r?\n'));
    final nodes = document.parseLines(lines);
```
and I can not get the `del` tag, I log it :

```
Tag:p   attr:{}
  Text:pas~~t~~
```
![image](https://user-images.githubusercontent.com/30992818/89707044-725cd200-d99d-11ea-9fbc-9ed80470d7a1.png)

If I add `space` before `~~` , it will be:

```
Tag:p   attr:{}
  Text:pas 
  Tag:del   attr:{}
    Text:t
```
![image](https://user-images.githubusercontent.com/30992818/89707054-856fa200-d99d-11ea-963e-1f876bbd391c.png)

**Comment by srawlins** [comment]
Interesting. The GitHub-Flavored Markdown is not as well-specified as vanilla CommonMark, but it does look like GFM parses `pas~~t~~` as struck-through text.

**Comment by asjqkkkk** [comment]
Hi, this issue seems still exist in v2.1.8

**Comment by srawlins** [comment]
Yes, it does. It will be fixed in 3.0.0
